test/common/tap-cdp

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python
+
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import print_function
+
+import argparse
+import cdp
+import subprocess
+import re
+import sys
+
+driver_line_re = re.compile(r'^> ([a-z]+): (.*)\n$')
+tap_line_re = re.compile(r'^(ok [0-9]+|not ok [0-9]+|bail out!|[0-9]+\.\.[0-9]+|# )', re.IGNORECASE)
+
+
+parser = argparse.ArgumentParser(description="A CDP driver for QUnit which outputs TAP")
+parser.add_argument("server", help="path to the test-server")
+parser.add_argument("test", help="the test page to run")
+
+# Strip prefix from url
+# We need this to compensate for automake test generation behavior:
+# The tests are called with the path (relative to the build directory) of the testfile,
+# but from the build directory. Some tests make assumptions regarding the structure of the
+# filename. In order to make sure that they receive the same name, regardless of actual
+# build directory location, we need to strip that prefix (path from build to source directory)
+# from the filename
+parser.add_argument("--strip", dest="strip", help="strip prefix from test file paths")
+
+opts = parser.parse_args()
+
+if opts.strip and opts.test.startswith(opts.strip):
+    opts.test = opts.test[len(opts.strip):]
+
+
+success = True
+
+server = subprocess.Popen([opts.server],
+                          stdout=subprocess.PIPE,
+                          stdin=subprocess.PIPE,
+                          close_fds=True)
+
+cdp = cdp.CDP("C.utf8")
+cdp.invoke("Page.navigate", url='file:///home/lars/src/cockpit/dist/docker/test-docker.html')
```

Comment:
I suppose this should be `opts.test` eventually? (I. e. this won't work on our CI like that)

Reply:
Oooops, sorry. I used this for testing and forgot to remove it.